### ISSUE TITLE
Terminal Issues 2

### ISSUE THREAD

**Issue by ALongStringOfNumbers** [GregTechCEu/GregTech#446]
The Second Round of Terminal Issues (after #158), of stuff that I noticed when playing around with the Terminal

### Widget Box

- When adding a Widget to a blank page in the App, it can be dragged off the blank page, which also messes with the scroll bar:
![2021-12-30_18-40](https://user-images.githubusercontent.com/31759736/147798227-eb8e3c8b-2e1a-4a75-a7b2-4117f1539d27.png)

They can even go under the widget selection screen:
![2021-12-30_18-49](https://user-images.githubusercontent.com/31759736/147798372-06bc6cfa-c36d-4b59-86a1-2942ea2da6e5.png)

Or all the way off the screen:
![2021-12-30_19-00](https://user-images.githubusercontent.com/31759736/147798721-e552b28e-99cc-469b-bd31-211d8a7de16c.png)

### Widget Config

- After adding a new "slot" widget and selecting the item for the slot, it would be nice if the item could be pulled from JEI bookmarks
- Selecting "Add to fixed" changes the widget selection type to whatever is under that button, when I would expect the widget selection type to not change.

## Page Config

- Clicking on the `type` box does not bring anything up, nor change anything on the page, and it does not have a hover tooltip, so I am not quite sure what it is for

## General

- After clicking on the "Load From File" button on the left popup, clicking on the directories in the file structure needs to be done twice (Once to select the folder and once again to open) to open subfolders, it would be nice if they only need to be clicked on once, aka combining select and expand

- After adding a "slots" Widget, you cannot click on the actual Item in the widget to select it, rather you have to click on the widget box outside of the item display. It would be nice to be able to click on the item to select the widget

- After selecting a "slot" widget, its L value displays under other "slot" widgets:
![2021-12-30_19-53](https://user-images.githubusercontent.com/31759736/147800368-1dba51b6-8ccf-468b-afb5-7d61de4dd5a2.png)

## Multiblock Helper App
- In AR mode, could a string be displayed across the bottom of the screen, telling users to "Right Click to exit App" (Same for the AR in the world prospector App)

## Tutorial App

- When opening this App and clicking on the "Resource Pack" Tab on the left to display information on the Emissive Textures, even though the title of the page is "Custom Emissive Textures", the title display in the left always starts at "Textures" and then cycles to "Custom Emissive", rather than starting at the beginning of the title. (This is the same as the entry for the Digital Interface cover in the Item Guides App)

- When searching for the string "test", this error came up in the log: https://pastebin.com/qpE6FAU7

- Search strings should be cleared between closing the app and re-opening it, and also closing the search box and re-opening it

## Hardware Manager App

- When the app is open, giving yourself the required hardware item (From JEI bookmarks or from JEI) does not properly show that the hardware is mountable when opening the hardware selection until you close the app and reopen it. Maybe should check for inventory change and then rescan for hardware items. 

## World Prospector App

- Does not find CEu machines correctly. I put a Creative Quantum Tank in the block selection, and nothing was found when using AR mode. I put an IV chemical reactor into the block selection, and all GT machines were found, including casing, creative energy emitters, and the creative tank.

- Highlighting in world is not in correct position. This might be related to the previous issue, as that is when I saw this.
With Highlighting:
![2021-12-30_21-12](https://user-images.githubusercontent.com/31759736/147802943-2525c8cc-9636-41f8-a9da-5ddfdf405aa4.png)

Without Highlighting:
![2021-12-30_21-12_1](https://user-images.githubusercontent.com/31759736/147802986-4ea8dd6d-5cbc-4898-a93e-d85d5b01a1fc.png)

Same place, I just scrolled off the terminal. In the first image, you can see that the blockspace above creative emitter is highlighted, when there is nothing there. As I said before though, this might be related to the previous issue, because this is when I was highlighting the chemical reactor from the previous issue.

Something similar in this:
![2021-12-30_21-31](https://user-images.githubusercontent.com/31759736/147803436-4d330aca-2d33-4ae0-b51c-d862f1f7d2e5.png)

It is probably just the algorithm used, trying to be performant

- App remains active when you scroll off the terminal. When you scroll off the terminal with the AR active, it remains active even though the terminal is no longer the active item. This can be seen by continuing power draw, even though the terminal is not the active item

- Slight bleed out of the highlight effect. At certain angles, the highlight effect can be seen outside the terminal item:
![2021-12-30_21-22](https://user-images.githubusercontent.com/31759736/147803214-2aad89fa-c205-4fc3-8ca9-fdb7426a6591.png)

![2021-12-30_21-23](https://user-images.githubusercontent.com/31759736/147803222-3c3f7aa2-399f-4a70-a650-0ae7b7b07d9b.png)

- Animation continues while game is paused. While the game is paused in singleplayer by hitting esc, the scan animation continues playing

## GT Console App

- Machine faces render invisible when color slider set to zero and face is obscured by another block
![2021-12-30_21-37](https://user-images.githubusercontent.com/31759736/147803661-b0edadcd-6308-4480-8cb4-e08ba049b664.png)

- Machine bottom faces render invisible when placed on a block and rotated to look at the bottom, on any slider level
![2021-12-30_21-40](https://user-images.githubusercontent.com/31759736/147803705-309076bf-782b-446c-a517-3c8b9a703f35.png)

- The "Set Working Enable" button should change its hover text based on the state of the machine. When the machine is not disabled, it should say "Set Working Disabled", when the machine is disabled, it should say "Set Working Enabled". Same with the "Allow Input from Output buttons

- As seen in #1012, there are culling issues when a cover is placed on a machine. This was tested with a solar cover.

## System Settings App

- When changing the Theme that affects App icon backgrounds, and the background in the System Settings App, the changes are not reflected until closing and reopening the terminal.

- It would be nice to have a hover tooltip over the different theme selector color bubbles, saying what they affect. I had no idea which color selector affected what

**Comment by pablocnvs** [comment]
As this seems the apropiate place, here's a crash log related to Multiblock debugging of the Fusion Reactor

[crash-2022-01-01_16.19.37-client.txt](https://github.com/GregTechCEu/GregTech/files/7797821/crash-2022-01-01_16.19.37-client.txt)

**Comment by serenibyss** [comment]
Going to assume these are either fixed, or minor enough that they are not an issue
